### ISSUE TITLE
`tril/triu` to have a `fill_value` option that defaults to zero.

### ISSUE THREAD

**Issue by nikitaved** [pytorch/pytorch#108347]
### 🚀 The feature, motivation and pitch

Sometimes it could be useful to return a lower/upper part of the matrix with the other part filled with something different than zero.
For example, if one wants to implement a scaled dot product attention with causality, then there is no need in materializing the attention mask and the operation could be implemented as
```
logits = (query @ key.transpose(-2, -1) * scale).triu_(fill_value=-float('inf'))
attention_weight = logits.softmax(dim=-1)
...
return attention_weight @ value
```

Seems like it is also very easy to implement.

cc @jianyuh @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

**Comment by daspartho** [comment]
i'd like to take on this!

**Comment by vadimkantorov** [comment]
Related:
- https://github.com/pytorch/pytorch/issues/97892

**Comment by nikitaved** [comment]
Thank you, @vadimkantorov , closing it then as duplicate. @daspartho , feel free to follow up in the https://github.com/pytorch/pytorch/issues/97892.
